Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+                                                         &fontFamilyExists));
+
+    if (!fontFamilyExists) {
+        TraceError(TAG, L"Unable to find font family \"%ws\"", unicharFamilyName.data());
+        return E_INVALIDARG;
+    }
+
+    return systemFontCollection->GetFontFamily(fontFamilyIndex, outFontFamily);
+}
+
+/**
+ * Helper function that creates an IDWriteFontFace object for a given font name.
+ */
+HRESULT _DWriteCreateFontFaceWithName(CFStringRef name, IDWriteFontFace** outFontFace) {
+    // Parse the font name for font weight, stretch, and style
+    // Eg: Bold, Condensed, Light, Italic
+    DWRITE_FONT_WEIGHT weight = DWRITE_FONT_WEIGHT_NORMAL;
+    DWRITE_FONT_STRETCH stretch = DWRITE_FONT_STRETCH_NORMAL;
+    DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+
+    CFStringRef familyName;
+    _InitDWriteFontPropertiesFromName(name, &weight, &stretch, &style, &familyName);
+
+    if (!familyName) {
+        TraceError(TAG, L"Unable to find family for font name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return E_INVALIDARG;
+    }
+
+    ComPtr<IDWriteFontFamily> fontFamily;
+
+    if (FAILED(_DWriteCreateFontFamilyWithName(familyName, &fontFamily)) || !fontFamily) {
+        TraceError(TAG, L"Unable to find family for font name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return E_INVALIDARG;
+    }
+
+    ComPtr<IDWriteFont> font;
+    HRESULT hr = fontFamily->GetFirstMatchingFont(weight, stretch, style, &font);
+    if (FAILED(hr)) {
+        TraceError(TAG, L"Unable to create font for name \"%hs\"", [static_cast<NSString*>(name) UTF8String]);
+        return hr;
+    }
+
+    return font->CreateFontFace(outFontFace);
+}
+
+/**
+ * Gets an informational string as a CFString from a DWrite font face.
+ */
+CFStringRef _DWriteFontCopyInformationalString(ComPtr<IDWriteFontFace> fontFace, DWRITE_INFORMATIONAL_STRING_ID informationalStringId) {
+    if (fontFace == nullptr) {
+        return nullptr;
+    }
+
+    ComPtr<IDWriteLocalizedStrings> name;
+    BOOL exists;
+
+    ComPtr<IDWriteFontFace3> dwriteFontFace3;
+    fontFace.As(&dwriteFontFace3);
+
+    THROW_IF_FAILED(dwriteFontFace3->GetInformationalStrings(informationalStringId, &name, &exists));
+    return exists ? static_cast<CFStringRef>(CFRetain(_CFStringFromLocalizedString(name.Get()))) : nullptr;
+}
+
+/**
+ * Gets a font table as a CFDataRef from a DWrite font face
+ */
+CFDataRef _DWriteFontCopyTable(ComPtr<IDWriteFontFace> fontFace, uint32_t tag) {
+    const void* tableData;
+    uint32_t tableSize;
+    void* tableContext;
+    BOOL exists;
+
+    if (FAILED(fontFace->TryGetFontTable(tag, &tableData, &tableSize, &tableContext, &exists)) || !exists) {
+        TraceError(TAG, L"Unable to get font table for tag %u", tag);
+        return nullptr;
+    }
+
+    // Copy the font table's binary data to a CFDataRef
+    CFDataRef ret = CFDataCreate(kCFAllocatorDefault, reinterpret_cast<const byte*>(tableData), tableSize);
```

Comment:
>CFDataRef ret [](start = 4, length = 13)

consider using a RAII friendly construct to avoid leaking this guy if an exception happens before its handed off. unique_cf is probably what you are looking for ...